importer/Monument.py

```diff
@@ -262,49 +258,70 @@ def create_wlm_source(self, monuments_all_id):
+
+    def exists_with_wd_item(self, wd_item_keyword="wd_item"):
+        """Check whether the data object has anything in the 'wd_item' field."""
+        if wd_item_keyword in self.raw_data:
+            return self.raw_data[wd_item_keyword]
+        else:
+            return None
+
     def exists_with_prop(self, mapping):
-        if self.existing is None:
-            return
+        """
+        Check whether there already exists an item
+        with the same unique ID as that in the data object.
+        """
         unique_prop = mapping.get_unique_prop()
         base = self.wd_item["statements"]
         if unique_prop in base:
             val_to_check = base[unique_prop][0]['value']
             if val_to_check in self.existing:
                 wd_item = self.existing[val_to_check]
-                print("Wikidata has item with {} = {}. Connecting with item {}.".format(
-                    unique_prop, val_to_check, wd_item))
-                # Check whether it already has self.wd_item
-                # from monument_article.
-                # This of course assumes that self.exist is run earlier.
-                # If it's a different one,
-                wd_item_from_monument_article = self.wd_item["wd-item"]
-                if wd_item != wd_item_from_monument_article:
-                    # different Q-numbers from monuments_all and
-                    # unique property
-                    self.wd_item["upload"] = False
-                else:
-                    pass  # match
-                self.set_wd_item(wd_item)
+                return wd_item
             else:
-                print("There's no item with {} = {} on Wikidata.".format(
-                    unique_prop, val_to_check))
+                return None
+
+    def in_known_items(self, wd_item):
+        """Check whether given item is on the list of items with certain prop"""
+        return wd_item in self.existing.items()
+
+    def find_matching_wikidata(self, mapping):
+        item = self.exists_with_prop(mapping)
+        if not item:
+            item = self.exists_with_wd_item()
+            if not item:
+                item = self.exists_with_monument_article(self.mapping["language"])
+            if item and self.in_known_items(item):
+                self.upload = False
+        return item
 
     def construct_wd_item(self, mapping, data_files=None):
+        self.mapping = mapping.file_content
         self.wd_item = {}
         self.wd_item["upload"] = True
         self.wd_item["statements"] = {}
         self.wd_item["labels"] = {}
         self.wd_item["aliases"] = {}
         self.wd_item["descriptions"] = {}
         self.wd_item["wd-item"] = None
-        self.mapping = mapping.file_content
 
     def print_datafiles_titles(self):
         """Print out the keys in self.data_files."""
         for title in self.data_files:
             print(title)
 
     def __init__(self, db_row_dict, mapping, data_files, existing):
```

Comment:
A docstring here defining the `existing` parameter (well all of them) would be good since we interact with it in `in_known_items()`